src/approaches/operator_learning_approach.py

```diff
@@ -36,9 +45,11 @@ def learn_from_offline_dataset(self, dataset: Dataset) -> None:
         for op in self._operators:
             # Sampler neural networks are already saved in operator_learning.py.
             # Here we'll save the other fields of the operator.
-            data.append((op.name, op.parameters, op.preconditions,
-                         op.add_effects, op.delete_effects, op.option_vars,
-                         op.option.name))
+            data.append((op.name, op.parameters,
+                         utils.atoms_to_pickleable(op.preconditions),
+                         utils.atoms_to_pickleable(op.add_effects),
```

Comment:
Hmm looking at this, I feel pretty strongly that we should figure out how to directly pickle things so we don't have to do conversions like this. Step one would probably be to convert the classifier functions into a class, rather than defining them as functions. I can give this a try if you want.